uikit/common/sdk/src/main/java/com/sygic/ui/common/sdk/mapobject/MapMarker.kt

```diff
@@ -0,0 +1,67 @@
+package com.sygic.ui.common.sdk.mapobject
+
+import android.graphics.Bitmap
+import androidx.annotation.DrawableRes
+import com.sygic.sdk.map.`object`.BitmapFactory
+import com.sygic.sdk.map.`object`.ViewObject
+import com.sygic.sdk.map.factory.DrawableFactory
+import com.sygic.sdk.map.factory.SimpleBitmapFactory
+import com.sygic.sdk.position.GeoCoordinates
+import com.sygic.ui.common.sdk.R
+import com.sygic.ui.common.sdk.data.PoiData
+
+private const val MARKER_ANCHOR_POSITION_X = 0.5f
+private const val MARKER_ANCHOR_POSITION_Y = 1f
+
+private val DEFAULT_ICON = DrawableFactory(R.drawable.ic_map_pin)
+
+// ToDo: Custom data / payload
+// ToDo: refactor SDK MapMarker, then remove it
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class MapMarker : com.sygic.sdk.map.`object`.MapMarker {
+
+    constructor(viewObject: ViewObject) : super(viewObject, DEFAULT_ICON) { this.data.coordinates = viewObject.position }
+    constructor(latitude: Double, longitude: Double) : super(GeoCoordinates(latitude, longitude), DEFAULT_ICON) { this.data.coordinates = GeoCoordinates(latitude, longitude) }
+    constructor(latitude: Double, longitude: Double, icon: BitmapFactory) : super(GeoCoordinates(latitude, longitude), icon) { this.data.coordinates = GeoCoordinates(latitude, longitude) }
+    constructor(latitude: Double, longitude: Double, icon: BitmapFactory, data: PoiData) : super(GeoCoordinates(latitude, longitude), icon) { this.data = data }
+
+    var data: PoiData = PoiData()
+
+    init {
+        setAnchorPosition(MARKER_ANCHOR_POSITION_X, MARKER_ANCHOR_POSITION_Y)
+    }
+
+    class Builder {
+
+        private val data: PoiData = PoiData()
+        private var bitmapFactory: BitmapFactory = DEFAULT_ICON
```

Comment:
tu tiez `lazy`

Reply:
ak nastavis vlastnu, ze by sa nemuselo toto inicializovat

Reply:
Tu to nejde, protože je to var :(

Reply:
nj :/